### PR TITLE
[CarPlay] Update bindings to Xcode 12.2 Beta 1

### PR DESCRIPTION
**Pull request by dalexsoto**
CPEntity docs got removed https://developer.apple.com/documentation/carplay/cpentity
so it is very likely that Apple will keep this out from the API.

### CODE REVIEW

File: src/CarPlay/CPCompat.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#if !XAMCORE_4_0
+namespace CarPlay {
+	[Register (SkipRegistration = true)]
+	[Introduced (PlatformName.iOS, 14, 0)]
+	[Unavailable (PlatformName.iOS, PlatformArchitecture.All)]
+	[Obsolete ("'CPEntity' has been removed from iOS.")]
+	[EditorBrowsable (EditorBrowsableState.Never)]
+	public class CPEntity : NSObject, INSSecureCoding {
+
+		public CPEntity () => throw new NotImplementedException ();
```
Comment (rolfbjarne) [inline src/CarPlay/CPCompat.cs:24]: I think we usually throw `NotSupportedException` instead of `NotImplementedException`.

**Review by rolfbjarne — APPROVED** [review]

File: src/CarPlay/CPCompat.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using Foundation;
+using ObjCRuntime;
+using System.ComponentModel;
+
+#if !XAMCORE_4_0
+namespace CarPlay {
+	[Register (SkipRegistration = true)]
+	[Introduced (PlatformName.iOS, 14, 0)]
+	[Unavailable (PlatformName.iOS, PlatformArchitecture.All)]
+	[Obsolete ("'CPEntity' has been removed from iOS.")]
```
Comment (spouliot) [inline src/CarPlay/CPCompat.cs:20]: we need to normalize those strings (to reduce the metadata size of our assemblies)
pick up an existing one, there's no need to mention the type name (or the OS), the IDE will put this in context

File: src/CarPlay/CPCompat.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	[EditorBrowsable (EditorBrowsableState.Never)]
+	public class CPEntity : NSObject, INSSecureCoding {
+
+		public CPEntity () => throw new NotImplementedException ();
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
```
Comment (spouliot) [inline src/CarPlay/CPCompat.cs:26]: there's already a `.Never` on the type
not sure what happens with the extra one - but `.Never` is the right one for obsolete code such as this

File: src/CarPlay/CPCompat.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		protected CPEntity (NSObjectFlag t) => throw new NotImplementedException ();
+
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
```
Comment (spouliot) [inline src/CarPlay/CPCompat.cs:29]: same

**Review by spouliot — CHANGES_REQUESTED** [review]

**Review by whitneyschmidt — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
The compat code should maintain the API that was public.

**Comment by dalexsoto** [comment]
@mandel-macaque ?? What am I missing here?

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**